src/tests/unit/tests/common/components/cards/instance-details.test.tsx

```diff
@@ -66,62 +73,70 @@ describe('InstanceDetails', () => {
             .setup(mock => mock.toggleCardSelection(It.isAnyString(), It.isAnyString(), It.isAny()))
             .verifiable(Times.never());
 
-        const testSubject = shallow(<InstanceDetails {...props} />);
+        const renderResult = render(<InstanceDetails {...props} />);
 
-        expect(testSubject.getElement()).toMatchSnapshot();
+        expect(renderResult.asFragment()).toMatchSnapshot();
         cardSelectionMessageCreatorMock.verifyAll();
     });
 
-    it('forwards focus and click events to hidden button when card is clicked', () => {
+    it('forwards focus and click events to hidden button when card is clicked', async () => {
         (React.useRef as any).mockReturnValueOnce(hiddenButtonRefStub);
         setupGetPropertyConfigByIdMock();
 
-        const wrapper = shallow(<InstanceDetails {...props} />);
-        const element = wrapper.find(`.${styles.instanceDetailsCard}`);
-        expect(element.length).toBe(1);
+        cardSelectionMessageCreatorMock
+            .setup(mock => mock.toggleCardSelection(It.isAnyString(), It.isAnyString(), It.isAny()))
+            .verifiable(Times.once());
+
+        const renderResult = render(<InstanceDetails {...props} />);
+        const element = renderResult.container.querySelector(`.${styles.instanceDetailsCard}`);
+        expect(element).toBeInTheDocument();
 
-        element.simulate('click');
+        fireEvent.click(element);
+        const button = renderResult.getByRole('button');
+        expect(button).toHaveFocus();
 
-        expect(hiddenButtonRefStub.current.focus).toHaveBeenCalled();
-        expect(hiddenButtonRefStub.current.click).toHaveBeenCalled();
+        cardSelectionMessageCreatorMock.verifyAll();
     });
 
-    it('dispatches the card selection message when hidden highlight button is clicked', () => {
+    it('dispatches the card selection message when hidden highlight button is clicked', async () => {
         (React.useRef as any).mockReturnValueOnce(hiddenButtonRefStub);
         setupGetPropertyConfigByIdMock();
         const stopPropagationMock = jest.fn();
-        const eventStub = {
-            stopPropagation: stopPropagationMock,
-        } as any;
 
         cardSelectionMessageCreatorMock
-            .setup(mock => mock.toggleCardSelection(It.isAnyString(), It.isAnyString(), eventStub))
+            .setup(mock => mock.toggleCardSelection(It.isAnyString(), It.isAnyString(), It.isAny()))
             .verifiable(Times.once());
 
-        const wrapper = shallow(<InstanceDetails {...props} />);
-        const element = wrapper.find(`.${styles.hiddenHighlightButton}`);
-        expect(element.length).toBe(1);
+        const renderResult = render(<InstanceDetails {...props} />);
+        const element = renderResult.container.querySelector(`.${styles.hiddenHighlightButton}`);
```

Comment:
```suggestion
        const element = renderResult.getByRole('button');
```

Reply:
Done, please review on the latest commit.